src/analysis/locally_convex/with_seminorms.lean

```diff
@@ -282,6 +303,16 @@ lemma seminorm_family.with_seminorms_of_has_basis (p : seminorm_family 𝕜 E ι
 p.with_seminorms_of_nhds $ filter.has_basis.eq_of_same_basis h
   p.add_group_filter_basis.to_filter_basis.has_basis
 
+lemma seminorm_family.with_seminorms_iff_nhds_eq_infi (p : seminorm_family 𝕜 E ι) :
+  with_seminorms p ↔ (𝓝 0 : filter E) = ⨅ i, (𝓝 0).comap (p i) :=
+begin
+  rw ← p.filter_eq_infi,
+  split,
+  { intro h,
+    rw h.topology_eq_with_seminorms,
+    exact add_group_filter_basis.nhds_zero_eq _ },
+  { exact p.with_seminorms_of_nhds }
```

Comment:
How about:
```suggestion
  rw ← p.filter_eq_infi,
  refine ⟨λ h, _, p.with_seminorms_of_nhds⟩,
  rw h.topology_eq_with_seminorms,
  exact add_group_filter_basis.nhds_zero_eq _,
```

Reply:
I see that you are forced to rewrite a typeclass because that's what the axiom `with_seminorms.topology_eq_with_seminorms` says.

I think we should avoid a design pattern that requires rewriting typeclasses so I think we should consider changing the definition of `with_seminorms` to something like:
```lean
class with_seminorms (p : seminorm_family 𝕜 E ι) [topological_space E] : Prop :=
(is_topological_family : ∀ (n ∈ 𝓝 (0 : E)), ∃ i ε, (p i).ball 0 ε ⊆ n)
```
(assuming I've got that right).

Of course this should be part of a future PR. This just seems like a good time to mention it. @mcdoll

Reply:
I'm not sure about this, currently in mathlib all the lattice operations on topological spaces are set up for rewriting and chaining (in)equalities about topologies, so it's better to have it stated like that I believe.

Reply:
Interesting, thanks @ADedecker perhaps this is an exceptional case!